frontend/src/views/CellGuide/components/CellGuideCard/components/MarkerGeneTables/index.tsx

```diff
@@ -191,234 +193,362 @@ export const MARKER_GENES_COMPUTATIONAL_TOOLTIP_TEST_ID =
-                        <i>{publicationTitle.split("\n\n").at(1)}</i>
-                      </div>
-                    }
-                    leaveDelay={0}
-                  >
-                    <span key={`${publications[index]}-${index}-span`}>
-                      <Link
-                        key={`${publications[index]}-${index}`}
-                        label={`[${referenceIndexLabel}]`}
-                        url={`https://doi.org/${publications[index]}`}
-                      />
-                    </span>
-                  </Tooltip>
-                );
-              }
-            })}
-          </PublicationLinkWrapper>
-        );
-
-        rows.push({
-          name,
-          symbolId: symbol,
-          symbol: (
-            <>
-              {symbol}{" "}
-              <ButtonIcon
-                aria-label={`display gene info for ${symbol}`}
-                sdsIcon="infoCircle"
-                sdsSize="small"
-                sdsType="secondary"
-                onClick={() => setGeneInfoGene(symbol.toUpperCase())}
-              />
-            </>
-          ),
-          references: publicationLinks,
-          numReferences: sortedPublicationsAndTitles.length,
-        });
-      }
+  useEffect(() => {
+    if (enrichedGenes) {
+      setComputationalMarkerGenes(enrichedGenes);
+    }
+  }, [enrichedGenes]);
 
-      // Sort rows by number of references
-      if (rows.length) {
-        rows.sort((a, b) => {
-          return b.numReferences - a.numReferences;
-        });
-      }
+  useEffect(() => {
+    if (canonicalMarkers) {
+      setCanonicalMarkerGenes(canonicalMarkers);
+    }
+  }, [canonicalMarkers]);
 
-      return rows;
-    }, [genes, selectedOrgan, setGeneInfoGene]);
-  }
+  const [page, setPage] = useState(1);
+
+  const {
+    uniqueOrganisms: uniqueOrganismsComputational,
+    uniqueOrgans: uniqueOrgansComputational,
+    tableRows: tableRowsComputational,
+    selectedOrganFilter: selectedOrganFilterComputational,
+    selectedOrganismFilter: selectedOrganismFilterComputational,
+  } = getComputationalMarkerGenesTableRowsAndFilters({
+    genes: computationalMarkerGenes,
+    selectedOrgan: selectedOrganComputational,
+    selectedOrganism: selectedOrganismComputational,
+    setGeneInfoGene,
+  });
+
+  const {
+    uniqueOrganisms: uniqueOrganismsCanonical,
+    uniqueOrgans: uniqueOrgansCanonical,
+    tableRows: tableRowsCanonical,
+    selectedOrganFilter: selectedOrganFilterCanonical,
+  } = getCanonicalMarkerGenesTableRowsAndFilters({
+    genes: canonicalMarkerGenes,
+    selectedOrgan: selectedOrganCanonical,
+    setGeneInfoGene,
+  });
+
+  const uniqueOrganisms = activeTable
+    ? uniqueOrganismsComputational
+    : uniqueOrganismsCanonical;
+  const uniqueOrgans = activeTable
+    ? uniqueOrgansComputational
+    : uniqueOrgansCanonical;
+  const tableRows = activeTable ? tableRowsComputational : tableRowsCanonical;
 
   useEffect(() => {
+    setPage(1);
```

Comment:
a neat way to reset a component state is to add the `key` prop to this component based on unique values you want to use to reset the component.

For example, in this case, we can remove this `useEffect()` with:

```tsx
// parent level
<MarkerGeneTables key={cellTypeId} />
```


https://medium.com/@albertogasparin/forcing-state-reset-on-a-react-component-by-using-the-key-prop-14b36cd7448e